### DISCUSSION TITLE
Ring Event issue

### DISCUSSION THREAD

**Discussion by Sebyrem** [tsightler/ring-mqtt#624] [Support Q&A]
HI team, I've just set up RING-MQTT with success, my camera have been added as generic camera, stream is working. But I've just discovered that enabling my ring camera as a generic will broke getting events from it even if i'v trio to get them on my original RING IOS APP. I mean without adding the camera as a generic one (with integration) events are working but as I integrate the camera as a generic one no more events ? what could be wrong on my setup ? thx for your help

**Comment by tsightler** [comment]
Ring cameras do not send events if there is a live stream, you must configure the camera to work only "on-demand", if you configure your camera to keep a stream active all the time, then you won't get any events.  This topic is covered in the documentation and the FAQ.

**Comment by Sebyrem** [comment]
Ok I do my best to understand but not so easy for me... OK so how can I use the stream link without adding it in a generic camera integration ? please teach me

**Reply by Sebyrem** [reply]
Doing my best I've tied disabling stream switch it works but the switch automatically go ON after 1 minutes ? can't find why ?

**Reply by tsightler** [reply]
Manually stopping the stream with the switch is almost always going to be useless unless you also started the stream with the switch (for recording puposes).  If you stop the stream, but any client connects to the stream, it will just start again, as the stream starts on-demand.

Did you configure the snapshot image in the generic camera config?  If not, then HA will always attempt to start a stream as HA is largely built to assume cameras can stream 24x7.  Also, you must disable "preload stream" and "live" modes in the HA UI or HA will constantly attempt to start a stream.

**Reply by Sebyrem** [reply]
When I configure the snapshot image url I get an error. My url is like https://192.168.1.XXX:8123{{ states.camera.porte_d'entrée_snapshot.attributes.entity_picture }}

Can you advice me ?

**Reply by tsightler** [reply]
You need to make sure the URL is correct and that the entitiy ID is correct.  I'm not sure if characters like ' are valid in a tempalte, they may need to be escaped.  The first part of the URL should be that same as you use to access your HA instance.

**Reply by Sebyrem** [reply]
So... I need help I've tried many snapshot url format without getting it... wo the snapshot the generic cam run but can't manage to get correct syntax ?

**Reply by Sebyrem** [reply]
So I finally get correct snapshot url but still the same as I add the generic cam to my Lovelace I get stream always activated I don't understand how to get it only on demand ?
